fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -668,24 +677,45 @@ public Resource doRead(String type, String id, boolean throwExcOnNull, boolean i
         return doRead(type, id, throwExcOnNull, includeDeleted, requestProperties, contextResource, null);
     }
 
+    @Override
+    public Resource doRead(String type, String id, boolean throwExcOnNull, boolean includeDeleted,
+            Map<String, String> requestProperties, Resource contextResource, MultivaluedMap<String, String> queryParameters)
+            throws Exception {
+        return doRead(type, id, throwExcOnNull, includeDeleted, requestProperties, contextResource, queryParameters, true);
+    }
+
     /**
      * Performs a 'read' operation to retrieve a Resource.
      *
      * @param type
      *            the resource type associated with the Resource to be retrieved
      * @param id
      *            the id of the Resource to be retrieved
+     * @param throwExcOnNull
+     *            if true, throw an exception if returned resource is null
+     * @param includeDeleted
+     *            if true, return resource even if deleted
      * @param requestProperties
      *            additional request properties which supplement the HTTP headers associated with this request
+     * @param contextResource
+     *            a FHIR resource associated with this request
+     * @param queryParameters
+     *            for supporting _summary for resource read
+     * @param doValidation
+     *            if true, validate the interaction; if false, assume the interaction does not need to be validated
```

Comment:
to me, it is a bit ambiguous as to whether this means it controls whether the resource will be validated or not (in addition to whether the interaction is allowed or not).

Reply:
Will change the parm name to checkInteractionAllowed.

Reply:
param name is changed